app/src/main-process/log.ts

```diff
@@ -78,12 +78,7 @@ function getLogger(): Promise<winston.LogMethod> {
 
     const logPath = getLogPath()
 
-    Fs.mkdir(logPath, (error) => {
-      if (error && error.code !== 'EEXIST') {
-        reject(error)
-        return
-      }
-
+    fileSystem.mkdirIfNeeded(logPath, (error) => { reject(error) }).then(() => {
```

Comment:
I thought the first function in the `then` callback was only called when the promise was fulfilled (see [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then))? so if it fails, the error callback would fire (and hence reject the wrapping promise in `log.ts`) but because there is only one arg passed to `then` it would only fire when the promise was fulfilled? I like the proposed change, just trying to clarify my understanding